python-sdk/src/astro/sql/operators/merge.py

```diff
@@ -73,8 +83,62 @@ def execute(self, context: Context) -> BaseTable:
             target_conflict_columns=self.target_conflict_columns,
             source_to_target_columns_map=self.columns,
         )
+        context["ti"].xcom_push(key="merge_query", value=str(db.sql))
         return self.target_table
 
+    def get_openlineage_facets(self, task_instance) -> OpenLineageFacets:
+        """
+        Collect the input, output, job and run facets for merge operator
+        """
+        input_dataset: list[OpenlineageDataset] = [
+            OpenlineageDataset(
+                namespace=self.source_table.openlineage_dataset_namespace(),
+                name=self.source_table.openlineage_dataset_name(),
+                facets={
+                    "input_table_facet": SourceTableMergeDatasetFacet(
+                        table_name=self.source_table.name,
+                        if_conflicts=self.if_conflicts,
+                        source_table_rows=self.source_table.row_count,
+                        columns=self.columns,
+                        metadata=self.source_table.metadata,
+                    ),
+                    "schema_dataset_facet": SchemaDatasetFacet(
+                        fields=[
+                            SchemaField(
+                                name=self.source_table.metadata.schema,
+                                type=self.source_table.metadata.database,
+                            )
+                        ]
+                    ),
+                },
+            )
+        ]
+
+        output_dataset: list[OpenlineageDataset] = [
+            OpenlineageDataset(
+                namespace=self.target_table.openlineage_dataset_namespace(),
+                name=self.target_table.openlineage_dataset_name(),
+                facets={
+                    "output_table_facet": TargetTableMergeDatasetFacet(
+                        table_name=self.target_table.name,
+                        target_conflict_columns=self.target_conflict_columns,
+                        columns=self.columns,
+                        metadata=self.target_table.metadata,
+                    ),
+                    "output_stats": OutputStatisticsOutputDatasetFacet(rowCount=self.target_table.row_count),
+                },
+            )
+        ]
+
+        run_facets: dict[str, BaseFacet] = {}
+
+        merge_query = task_instance.xcom_pull(task_ids=task_instance.task_id, key="merge_query")
+        job_facets: dict[str, BaseFacet] = {"sql": SqlJobFacet(query=str(merge_query))}
```

Comment:
@sunank200 do we need typecast `str` since we are pushing str in xcom?

Reply:
I can remove this